dm/master/agent_pool.go

```diff
@@ -14,20 +14,38 @@
 package master
 
 import (
+	"context"
+	"math"
 	"sync"
+
+	"golang.org/x/time/rate"
 )
 
 var (
-	pool       *AgentPool // singleton instance
-	once       sync.Once
-	agentlimit = 20
+	pool             *AgentPool // singleton instance
+	once             sync.Once
+	defalutRate      float64 = 10
+	defaultBurst             = 40
+	errorNoEmitToken         = "fail to get emit opporunity for %s"
 )
 
+type emitFunc func(args ...interface{})
```

Comment:
can we export `emitFunc` and `errorNoEmitToken`?

Reply:
`errorNoEmitToken` exposed. `emitFunc` seems not directly used by external module